ex-2/client/client.c

```diff
@@ -0,0 +1,123 @@
+#include <stdio.h>
+#include <sys/socket.h>
+#include <errno.h>
+#include <netinet/in.h>
+#include <unistd.h>
+
+#define BSIZE 1024
+#define PORT 21996
+
+void CleanStdInput(){
+    while(fgetc(stdin) != 10);
+}
+
+void GetStringFromStdInput(char* s, unsigned int length){
+	fgets(s, length, stdin);
+	*(s + strlen(s) - 1) = 0;
+}
+
+enum DATA_TYPE{
+    NUMBER, STRING, STUDENT_INFO
+};
+
+
+struct Student{
+    char id[10];
+    char name[100];
+    struct {
+        unsigned int day, month;
+        unsigned int year;
+    } birthday;
+};
+
+
+int main(){
+    char buff[BSIZE];
+    int stop = 0;
+    int type;
+    int cskid = socket(AF_INET, SOCK_STREAM, 0);
+    int sbytes;
+    struct Student student;
+
+    int number;
+
+    if (cskid == -1){
+        perror("Create client socket failed");
+        return 0x0;
+    }
+
+    struct sockaddr_in serveraddr = {
+            AF_INET,
+            htons(PORT),
+            htonl(0x7f000001)
+    };
+
+    if (connect(cskid, (struct sockaddr*)&serveraddr, sizeof serveraddr) == -1){
+        perror("Connect to server failed");
```

Comment:
Why don't we free the socket and exit here, when the connect failed?

Reply:
Line 120, I free it.

Reply:
Sorry RET main before free it.